### PR TITLE
(Docs) Updates RBAC actions for Adaptive Metrics

### PR DESCRIPTION
**Pull request by tacole02**
**What is this feature?**

This PR updates https://grafana.com/docs/grafana/latest/administration/roles-and-permissions/access-control/custom-role-actions-scopes/ to include custom RBAC actions for Adaptive Metrics

**Why do we need this feature?**

Many Adaptive Metrics customers are unaware that they can use RBAC to manage permissions to Adaptive Metrics features.

**Who is this feature for?**

Adaptive Metrics customers.

**Which issue(s) does this PR fix?**:

https://github.com/grafana/grafana-adaptive-metrics-app/issues/520

Please check that:
- [X] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [X] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Comment by tacole02** [comment]
Because tables in markdown don't look nice, here's a screenshot:
image

**Review by nfuzer — CHANGES_REQUESTED** [review]
Hi Taylor, 

For this issue, I was going to suggest reaching out to @xiyu95 to get the complete list of "action" descriptions for Adaptive Metrics. Also Xiyu, if you happen to see this comment, please inform Taylor which ones are missing. 

Based on what we have in the plugin under "teams" and "users" we have these Plugin Roles: Screenshot 2024-07-24 at 5 49 11 PM

My request is:  
(1) document the complete list of actions including editor, reader, access, and admin permissions 
(2) open up a second PR to document the roles as they are stated in the product, which can live in the Adaptive Metrics doc, e.g. (Admin, Config Editor, Config Reader, Exemptions Editor, Exemptions Reader, Plugin Access, Rules Editor, Rules Reader)   

Also since we haven't had a docs writer or copy editor for the naming convention in the plugin, if you have any suggestions or something seems awkward to you about how we are naming anything, pleaes let me know!  Thanks!

**Comment by tacole02** [comment]
From @xiyu95 , here is the list of supported actions:
```
export type PluginPermissions =
  | 'grafana-adaptive-metrics-app.auto-apply:read'
  | 'grafana-adaptive-metrics-app.auto-apply:write'
  | 'grafana-adaptive-metrics-app.exemptions:read'
  | 'grafana-adaptive-metrics-app.exemptions:write'
  | 'grafana-adaptive-metrics-app.plugin:access'
  | 'grafana-adaptive-metrics-app.recommendations:apply'
  | 'grafana-adaptive-metrics-app.recommendations:read'
  | 'grafana-adaptive-metrics-app.rules:delete'
  | 'grafana-adaptive-metrics-app.rules:read'
  | 'grafana-adaptive-metrics-app.rules:write'
  | 'grafana-adaptive-metrics-app.config:read'
  | 'grafana-adaptive-metrics-app.config:write'
```
And here are the default role definitions:
```
"roles": [
    {
      "role": {
        "name": "Admin",
        "description": "Read/write access to everything in Adaptive Metrics",
        "permissions": [
          { "action": "grafana-adaptive-metrics-app.plugin:access"},

          { "action": "grafana-adaptive-metrics-app.recommendations:read" },
          { "action": "grafana-adaptive-metrics-app.recommendations:apply" },

          { "action": "grafana-adaptive-metrics-app.rules:read" },
          { "action": "grafana-adaptive-metrics-app.rules:write" },
          { "action": "grafana-adaptive-metrics-app.rules:delete" },

          { "action": "grafana-adaptive-metrics-app.exemptions:read" },
          { "action": "grafana-adaptive-metrics-app.exemptions:write" },

          { "action": "grafana-adaptive-metrics-app.config:read" },
          { "action": "grafana-adaptive-metrics-app.config:write" }
        ]
      },
      "grants": ["Grafana Admin", "Admin"]
    },
    {
      "role": {
        "name": "Rules Editor",
        "description": "Read/write access to recommendations and rules in Adaptive Metrics",
        "permissions": [
          { "action": "grafana-adaptive-metrics-app.plugin:access"},

          { "action": "grafana-adaptive-metrics-app.recommendations:read" },
          { "action": "grafana-adaptive-metrics-app.recommendations:apply" },

          { "action": "grafana-adaptive-metrics-app.rules:read" },
          { "action": "grafana-adaptive-metrics-app.rules:write" },
          { "action": "grafana-adaptive-metrics-app.rules:delete" }
        ]
      },
      "grants": []
    },
    {
      "role": {
        "name": "Rules Reader",
        "description": "Read access to recommendations and rules in Adaptive Metrics",
        "permissions": [
          { "action": "grafana-adaptive-metrics-app.plugin:access"},

          { "action": "grafana-adaptive-metrics-app.recommendations:read" },

          { "action": "grafana-adaptive-metrics-app.rules:read" }
        ]
      },
      "grants": []
    },
    {
      "role": {
        "name": "Exemptions Editor",
        "description": "Edit access to recommendation exemptions in Adaptive Metrics",
        "permissions": [
          { "action": "grafana-adaptive-metrics-app.plugin:access"},

          { "action": "grafana-adaptive-metrics-app.exemptions:read" },
          { "action": "grafana-adaptive-metrics-app.exemptions:write" }
        ]
      },
      "grants": []
    },
    {
      "role": {
        "name": "Exemptions Reader",
        "description": "Read access to recommendation exemptions in Adaptive Metrics",
        "permissions": [
          { "action": "grafana-adaptive-metrics-app.plugin:access"},

          { "action": "grafana-adaptive-metrics-app.exemptions:read" }
        ]
      },
      "grants": []
    },
    {
      "role": {
        "name": "Config Editor",
        "description": "Edit access to Adaptive Metrics plugin config",
        "permissions": [
          { "action": "grafana-adaptive-metrics-app.plugin:access"},

          { "action": "grafana-adaptive-metrics-app.config:read" },
          { "action": "grafana-adaptive-metrics-app.config:write" }
        ]
      },
      "grants": []
    },
    {
      "role": {
        "name": "Config Reader",
        "description": "Read access to Adaptive Metrics plugin config",
        "permissions": [
          { "action": "grafana-adaptive-metrics-app.plugin:access"},

          { "action": "grafana-adaptive-metrics-app.config:read" }
        ]
      },
      "grants": []
    },
    {
      "role": {
        "name": "Plugin Access",
        "description": "Access to the Adaptive Metrics plugin",
        "permissions": [
          { "action": "plugins.app:access", "scope": "plugins:id:grafana-adaptive-metrics-app" },
          { "action": "grafana-adaptive-metrics-app.plugin:access"}
        ]
      },
      "grants": []
    }
  ],
```

File: docs/sources/administration/roles-and-permissions/access-control/custom-role-actions-scopes/index.md [hunk 1]
```diff
@@ -205,6 +205,23 @@ The following list contains role-based access control actions used by Grafana On
...
+| grafana-adaptive-metrics-app.config:read           | n/a              | Read the Adaptive Metrics app configuration.         |
+| grafana-adaptive-metrics-app.config:write          | n/a              | Update the Adaptive Metrics app configuration.       |
+| grafana-adaptive-metrics-app.recommendations:read  | n/a              | Read aggregation recommendations.                    |
+| grafana-adaptive-metrics-app.recommendations:apply | n/a              | Apply aggregation recommendations.                   |
+| grafana-adaptive-metrics-app.rules:read            | n/a              | Read aggregation rules.                              |
+| grafana-adaptive-metrics-app.rules:write           | n/a              | Create aggregation rules.                            |
+| grafana-adaptive-metrics-app.rules:delete          | n/a              | Delete aggregation rules.                            |
+| grafana-adaptive-metrics-app.exemptions:read       | n/a              | Read rule exemptions.                                |
+| grafana-adaptive-metrics-app.exemptions:write      | n/a              | Create rule exemptions.                              |
```
Comment (xiyu95) [inline docs/sources/administration/roles-and-permissions/access-control/custom-role-actions-scopes/index.md:223]: I think the `write` isn't necessary just create, they are able to edit/remove as well.

Comment (tacole02) [reply docs/sources/administration/roles-and-permissions/access-control/custom-role-actions-scopes/index.md:223]: Thanks, @xiyu95 , what if we changed the description for `grafana-adaptive-metrics-app.exemptions:write`  to `Create, update, and delete rule exemptions.`?

**Comment by tacole02** [comment]
I've updated the list of supported actions in the Grafana RBAC docs. I'll complete a second PR for the Adaptive Metrics docs following this merge.

image

File: docs/sources/administration/roles-and-permissions/access-control/custom-role-actions-scopes/index.md [hunk 2]
```diff
@@ -205,6 +205,23 @@ The following list contains role-based access control actions used by Grafana On
...
+| grafana-adaptive-metrics-app.config:read           | n/a              | Read the Adaptive Metrics app configuration.         |
+| grafana-adaptive-metrics-app.config:write          | n/a              | Update the Adaptive Metrics app configuration.       |
+| grafana-adaptive-metrics-app.recommendations:read  | n/a              | Read aggregation recommendations.                    |
+| grafana-adaptive-metrics-app.recommendations:apply | n/a              | Apply aggregation recommendations.                   |
+| grafana-adaptive-metrics-app.rules:read            | n/a              | Read aggregation rules.                              |
+| grafana-adaptive-metrics-app.rules:write           | n/a              | Create aggregation rules.                            |
+| grafana-adaptive-metrics-app.rules:delete          | n/a              | Delete aggregation rules.                            |
+| grafana-adaptive-metrics-app.exemptions:read       | n/a              | Read rule exemptions.                                |
+| grafana-adaptive-metrics-app.exemptions:write      | n/a              | Create, update, and delete rule exemptions.          |
```
Comment (jdbaldry) [inline docs/sources/administration/roles-and-permissions/access-control/custom-role-actions-scopes/index.md:223]: I think to be consistent with the preceding table, all the Actions should be in code? https://grafana.com/docs/writers-toolkit/write/style-guide/style-conventions/#code

**Review by jdbaldry — APPROVED** [review]
LGTM, one nit

**Review by nfuzer — CHANGES_REQUESTED** [review]
instead of "rule exemptions" we refer to them as "recommendation exemptions"

**Review by nfuzer — APPROVED** [review]

**Pull request merged**